pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/channel/ChannelManager.java

```diff
@@ -38,30 +43,62 @@
+   * period of inactivity.
+   *
+   * This is why by default we set the idle timeout to twice the pinger period if a pinger is configured, so that the
+   * pinger can keep the channel alive. In case the pinger is not configured, we set the idle timeout to 30 minutes,
+   * which is the default value in the gRPC Java implementation.
+   */
+  private final Duration _idleTimeout;
 
-  public ChannelManager(@Nullable TlsConfig tlsConfig) {
+  public ChannelManager(@Nullable TlsConfig tlsConfig, Duration idleTimeout) {
+    Preconditions.checkArgument(idleTimeout.isNegative() || idleTimeout.isZero(), "Idle timeout must be positive");
     _tlsConfig = tlsConfig;
+    _idleTimeout = idleTimeout;
   }
 
   public ManagedChannel getChannel(String hostname, int port) {
     // TODO: Revisit parameters
     if (_tlsConfig != null) {
       return _channelMap.computeIfAbsent(Pair.of(hostname, port),
-          (k) -> NettyChannelBuilder
-              .forAddress(k.getLeft(), k.getRight())
-              .maxInboundMessageSize(
-                  CommonConstants.MultiStageQueryRunner.DEFAULT_MAX_INBOUND_QUERY_DATA_BLOCK_SIZE_BYTES)
-              .sslContext(ServerGrpcQueryClient.buildSslContext(_tlsConfig))
-              .build()
+          (k) -> {
+            NettyChannelBuilder channelBuilder = NettyChannelBuilder
+                .forAddress(k.getLeft(), k.getRight())
+                .maxInboundMessageSize(
+                    CommonConstants.MultiStageQueryRunner.DEFAULT_MAX_INBOUND_QUERY_DATA_BLOCK_SIZE_BYTES)
+                .sslContext(ServerGrpcQueryClient.buildSslContext(_tlsConfig));
+            return decorate(channelBuilder).build();
+          }
       );
     } else {
       return _channelMap.computeIfAbsent(Pair.of(hostname, port),
-          (k) -> ManagedChannelBuilder
-              .forAddress(k.getLeft(), k.getRight())
-              .maxInboundMessageSize(
-                  CommonConstants.MultiStageQueryRunner.DEFAULT_MAX_INBOUND_QUERY_DATA_BLOCK_SIZE_BYTES)
-              .usePlaintext()
-              .build());
+          (k) -> {
+            ManagedChannelBuilder<?> channelBuilder = ManagedChannelBuilder
+                .forAddress(k.getLeft(), k.getRight())
+                .maxInboundMessageSize(
+                    CommonConstants.MultiStageQueryRunner.DEFAULT_MAX_INBOUND_QUERY_DATA_BLOCK_SIZE_BYTES)
+                .usePlaintext();
+            return decorate(channelBuilder).build();
+          });
     }
   }
+
+  private ManagedChannelBuilder<?> decorate(ManagedChannelBuilder<?> builder) {
+    return builder.idleTimeout(_idleTimeout.getSeconds(), TimeUnit.SECONDS);
+  }
+
+  /**
+   * Returns a set with the key of all known channels.
+   *
+   * The returned value is a copy of the internal map's key set, so it is safe to use
+   * without worrying about concurrent modifications.
+   */
+  public Set<Pair<String, Integer>> getKnownChannels() {
```

Comment:
nit: might be worth documenting what a `key` is here since it's a public method.